pkg/storage/chunk/chunk_store_utils.go

```diff
@@ -16,6 +19,23 @@ import (
 	"github.com/grafana/loki/pkg/storage/chunk/cache"
 )
 
+var (
+	errAsyncBufferFull    = errors.New("the async buffer is full")
+	reasonAsyncBufferFull = "async-buffer-full"
+	skipped               = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "loki_chunk_fetcher_cache_skipped_buffer_full_total",
+		Help: "Total number of operations against cache that have been skipped.",
+	}, []string{"reason"})
+	chunkFetcherCacheQueueEnqueue = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "loki_chunk_fetcher_cache_enqueue_total",
+		Help: "Total number of chunk enqueue cache queue.",
+	})
+	chunkFetcherCacheQueueDequeue = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "loki_chunk_fetcher_cache_dequeue_total",
+		Help: "Total number of chunk dequeue cache queue.",
+	})
```

Comment:
Minor wording suggestions

```suggestion
	chunkFetcherCacheQueueEnqueue = promauto.NewCounter(prometheus.CounterOpts{
		Name: "loki_chunk_fetcher_cache_enqueued_total",
		Help: "Total number of chunks enqueued to a buffer to be asynchronously written back to the chunk cache.",
	})
	chunkFetcherCacheQueueDequeue = promauto.NewCounter(prometheus.CounterOpts{
		Name: "loki_chunk_fetcher_cache_dequeued_total",
		Help: "Total number of chunks asynchronously dequeued from a buffer and written back to the chunk cache.",
	})
```